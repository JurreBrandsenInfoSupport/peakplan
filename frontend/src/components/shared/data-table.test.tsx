import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { DataTable } from './data-table';
import { ColumnDef } from '@tanstack/react-table';

describe('DataTable', () => {
    // Define test types
    type TestData = {
        id: number;
        name: string;
        email: string;
    };

    // Define test columns
    const columns: ColumnDef<TestData>[] = [
        {
            accessorKey: 'id',
            header: 'ID',
        },
        {
            accessorKey: 'name',
            header: 'Name',
        },
        {
            accessorKey: 'email',
            header: 'Email',
            meta: {
                className: 'hidden md:table-cell',
            },
        },
    ];

    // Define test data
    const data: TestData[] = [
        { id: 1, name: 'John Doe', email: 'john@example.com' },
        { id: 2, name: 'Jane Smith', email: 'jane@example.com' },
    ];

    // Helper function to render the component
    const renderDataTable = () => {
        return render(<DataTable columns={columns} data={data} />);
    };

    it('renders the table with data correctly', () => {
        const component = renderDataTable();
        expect(component).toMatchSnapshot();
    });

    it('renders empty state when no data is provided', () => {
        const component = render(<DataTable columns={columns} data={[]} />);
        expect(component).toMatchSnapshot();
    });

    it('renders with custom column classes', () => {
        const component = renderDataTable();
        expect(component).toMatchSnapshot();
    });
});