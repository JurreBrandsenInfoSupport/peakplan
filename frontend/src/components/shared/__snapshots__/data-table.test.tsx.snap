// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTable > renders empty state when no data is provided 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-md border"
      >
        <div
          class="relative w-full overflow-x-auto"
          data-slot="table-container"
        >
          <table
            class="w-full caption-bottom text-sm"
            data-slot="table"
          >
            <thead
              class="[&_tr]:border-b"
              data-slot="table-header"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
              >
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  ID
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Name
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] hidden md:table-cell"
                  data-slot="table-head"
                >
                  Email
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
              data-slot="table-body"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
              >
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] h-24 text-center"
                  colspan="3"
                  data-slot="table-cell"
                >
                  No results.
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-md border"
    >
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                ID
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Name
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] hidden md:table-cell"
                data-slot="table-head"
              >
                Email
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] h-24 text-center"
                colspan="3"
                data-slot="table-cell"
              >
                No results.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DataTable > renders the table with data correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-md border"
      >
        <div
          class="relative w-full overflow-x-auto"
          data-slot="table-container"
        >
          <table
            class="w-full caption-bottom text-sm"
            data-slot="table"
          >
            <thead
              class="[&_tr]:border-b"
              data-slot="table-header"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
              >
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  ID
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Name
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] hidden md:table-cell"
                  data-slot="table-head"
                >
                  Email
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
              data-slot="table-body"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
                data-state="false"
              >
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  1
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  John Doe
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  john@example.com
                </td>
              </tr>
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
                data-state="false"
              >
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  2
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  Jane Smith
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  jane@example.com
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-md border"
    >
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                ID
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Name
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] hidden md:table-cell"
                data-slot="table-head"
              >
                Email
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
              data-state="false"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                1
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                John Doe
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                john@example.com
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
              data-state="false"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                2
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Jane Smith
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                jane@example.com
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DataTable > renders with custom column classes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-md border"
      >
        <div
          class="relative w-full overflow-x-auto"
          data-slot="table-container"
        >
          <table
            class="w-full caption-bottom text-sm"
            data-slot="table"
          >
            <thead
              class="[&_tr]:border-b"
              data-slot="table-header"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
              >
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  ID
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Name
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] hidden md:table-cell"
                  data-slot="table-head"
                >
                  Email
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
              data-slot="table-body"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
                data-state="false"
              >
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  1
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  John Doe
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  john@example.com
                </td>
              </tr>
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
                data-state="false"
              >
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  2
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  Jane Smith
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-cell"
                >
                  jane@example.com
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-md border"
    >
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                ID
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Name
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] hidden md:table-cell"
                data-slot="table-head"
              >
                Email
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
              data-state="false"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                1
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                John Doe
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                john@example.com
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
              data-state="false"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                2
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Jane Smith
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                jane@example.com
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
